### PR TITLE
Vibhansa/msi pipeline adls

### PR DESCRIPTION
**Pull request by vibhansa-msft**
Adding Block blob and ADLS MSI auth test.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ For more information, see the [wiki](https://github.com/Azure/azure-storage-fuse
 
 - General options
     * `accountName`: Specifies the storage account blobfuse targets.
-    * `blobEndpoint`: Specifies the blob endpoint to use. Defaults to *.blob.core.windows.net, but is useful for targeting storage emulators.
+    * `blobEndpoint`: Specifies the blob endpoint to use. Defaults to *.blob.core.windows.net, but is useful for targeting storage emulators. (e.g. storageaccountname.blob.region.azurestack.company.com)
```
Comment (NaraVen) [inline README.md:116]: the writing should " Optional parameter, if it is not specified defaults to blob.core.windows.net". This parameter should be specified for zonal dns accounts,  custom domain name accounts or storage emulators.

File: blobfuse/blobfuse.cpp [hunk 2]
```diff
@@ -327,28 +341,45 @@ void *azs_init(struct fuse_conn_info * conn)
...
+            char errStr[] = "Unable to start blobfuse due to a lack of credentials. Please check the readme for valid auth setups.\n";
+            if (stdErrFD != -1) {
+                ssize_t n = write(stdErrFD, errStr, sizeof(errStr));
+                if (n == -1) {
+                    syslog(LOG_ERR, "Failed to report back the status of auth.");
+                }
+            }
+            syslog(LOG_ERR, "%s", errStr);
+            std::thread t1(std::bind(&destroyBlobfuseOnAuthError));
+            t1.detach();
```
Comment (NaraVen) [inline blobfuse/blobfuse.cpp:369]: does this detach succeed even if the main thread is executing? In my testing if I am in that dir it did not unmount.
Can we throw an error if unmount fails so that the user knows they have to unmount manually?

File: blobfuse/blobfuse.cpp [hunk 3]
```diff
@@ -304,6 +306,18 @@ int read_config(const std::string configFile)
...
+    if (n == -1) {
+        syslog(LOG_ERR, "Failed to report back failure.");
+    }
+    close(stdErrFD);
+    fuse_unmount(config_options.mntPath.c_str(), NULL);
+}
+
 int configure_tls();
 void *azs_init(struct fuse_conn_info * conn)
 {
```
Comment (NaraVen) [inline blobfuse/blobfuse.cpp:323]: kernel_version < 4.16 can be replaced by blobfuse_constants.min_kernel_version or equivalent

**Review by NaraVen — CHANGES_REQUESTED** [review]

File: blobfuse/blobfuse.cpp (hunk 3 above)
Comment (vibhansa-msft) [reply blobfuse/blobfuse.cpp:323]: Will change accordingly.

File: blobfuse/blobfuse.cpp (hunk 2 above)
Comment (vibhansa-msft) [reply blobfuse/blobfuse.cpp:369]: fuse_unmount is of void return type so we need to explicitly check whether unmount was successful or not. By the call itself we can not judge it.

Comment (vibhansa-msft) [reply blobfuse/blobfuse.cpp:390]: Have added mount directory check post unmount for the error reporting.

**Review by NaraVen — APPROVED** [review]
